Don't use -arch flags when building gen_manual. It will be run on the
build machine so it needs to be built for the build machine's arch.
--- contrib/gen_manual/Makefile.orig	2020-11-15 22:59:35.000000000 -0600
+++ contrib/gen_manual/Makefile	2021-01-12 00:24:35.000000000 -0600
@@ -34,6 +34,7 @@
 CXXFLAGS += -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wno-comment
 CXXFLAGS += $(MOREFLAGS)
 FLAGS   = $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS)
+FLAGS := $(filter-out -arch^%,$(subst -arch ,-arch^,$(FLAGS)))
 
 LZ4API = ../../lib/lz4.h
 LZ4MANUAL = ../../doc/lz4_manual.html
